src/lib/multicall/executeMulticallWorker.ts

```diff
@@ -35,9 +37,8 @@ export async function executeMulticallWorker(chainId: number, request: any) {
     PRODUCTION_PREVIEW_KEY: localStorage.getItem(PRODUCTION_PREVIEW_KEY),
   });
 
-  const promise = new Promise((resolve, reject) => {
-    promises.set(id, { resolve, reject });
-  });
+  const { promise, resolve, reject } = Promise.withResolvers();
+  promises[id] = { resolve, reject };
```

Comment:
sorry, small comment here, let's maybe do some timeout with reject here?
for example 
```
setTimeout(() => reject(new Error("Worker stuck"), 3000);
```

and same in hashing with lower timeout. maybe 1 second?

Reply:
agree, done 3 seconds